### DISCUSSION TITLE
Windicss in gulp workflow

### DISCUSSION THREAD

**Discussion by DJI1957** [windicss/windicss#188] [Q&A]
Hi, I would like to replace tailwindcss with windi in a gulp workflow, can anyone advise how the gulpfile.js should look. Thanks

**Comment by DJI1957** [comment]
More specifically
function devStyles(){
const windicss = require('windicss');
  return src(`${options.paths.src.css}/**/*`).pipe(sass().on('error', sass.logError))
    .pipe(windicss(options.config.windijs)
      )
    .pipe(concat({ path: 'style.css'}))
    .pipe(dest(options.paths.dist.css));
}
generates the following error:-
TypeError: Cannot set property '_variants' of undefined
    at Processor (/data/winditest/windivm/public/node_modules/windicss/lib/index.js:8293:24)

**Comment by antfu** [comment]
Hi @DJI1957, I am not familiar with gulpfile but I'd love to help with that. Can you provide a repo with the app setup so I can play with it and see if we can find out a solution to make a gulp plugin? Thanks

**Reply by DJI1957** [reply]
Hi , thanks for the response, I'll try and get that set up, will let you know

**Reply by DJI1957** [reply]
To save time, I would like to modify this repo lazymozek/gulp-with-tailwindcss to use windicss if that helps at all

**Comment by antfu** [comment]
Made a PostCSS plugin, maybe you can try with it first
https://github.com/windicss/postcss-windicss

**Reply by DJI1957** [reply]
Thanks a lot, I'll try it in the next day or so
